x-pack/plugins/case/common/api/connectors/index.ts

```diff
@@ -3,7 +3,80 @@
  * or more contributor license agreements. Licensed under the Elastic License;
  * you may not use this file except in compliance with the Elastic License.
  */
+import * as rt from 'io-ts';
+
+import { JiraFieldsRT } from './jira';
+import { ResilientFieldsRT } from './resilient';
+import { ServiceNowFieldsRT } from './servicenow';
 
 export * from './jira';
 export * from './servicenow';
 export * from './resilient';
+
+export const ConnectorFieldsRt = rt.union([
+  JiraFieldsRT,
+  ResilientFieldsRT,
+  ServiceNowFieldsRT,
+  rt.null,
+]);
+
+export enum ConnectorTypes {
+  jira = '.jira',
+  resilient = '.resilient',
+  servicenow = '.servicenow',
+  none = '.none',
+}
+
+const ConnectorJiraTypeFieldsRt = rt.type({
+  type: rt.literal(ConnectorTypes.jira),
+  fields: rt.union([JiraFieldsRT, rt.null]),
+});
+
+const ConnectorResillientTypeFieldsRt = rt.type({
+  type: rt.literal(ConnectorTypes.resilient),
+  fields: rt.union([ResilientFieldsRT, rt.null]),
+});
+
+const ConnectorServiceNowTypeFieldsRt = rt.type({
+  type: rt.literal(ConnectorTypes.servicenow),
+  fields: rt.union([ServiceNowFieldsRT, rt.null]),
+});
+
+const ConnectorNoneTypeFieldsRt = rt.type({
+  type: rt.literal(ConnectorTypes.none),
+  fields: rt.null,
+});
+
+export const ConnectorTypeFieldsRt = rt.union([
+  ConnectorJiraTypeFieldsRt,
+  ConnectorResillientTypeFieldsRt,
+  ConnectorServiceNowTypeFieldsRt,
+  ConnectorNoneTypeFieldsRt,
+]);
+
+export const CaseConnectorRt = rt.intersection([
+  rt.type({
+    id: rt.string,
+    name: rt.string,
+  }),
+  ConnectorTypeFieldsRt,
+]);
+
+export type CaseConnector = rt.TypeOf<typeof CaseConnectorRt>;
+export type ConnectorTypeFields = rt.TypeOf<typeof ConnectorTypeFieldsRt>;
+
+// we need to change these types back and forth for storing in ES (arrays overwrite, objects merge)
+export type ConnectorFields = rt.TypeOf<typeof ConnectorFieldsRt>;
+
+export type ESConnectorFields = Array<{
+  key: string;
+  value: unknown;
+}>;
+
+export type ESCaseConnectorTypes = ConnectorTypes;
```

Comment:
why can we just use `ConnectorTypes` if there is no difference between `ESCaseConnectorTypes` and `ConnectorTypes`. I just feel like it is getting confusing and hard to follow when there is tow type doing the same thing.

Reply:
We will do that in a follow-up PR